specs/merge/fork-choice.md

```diff
@@ -43,25 +43,13 @@ The body of this function is implementation dependent.
 The Consensus API may be used to implement this with an external execution engine.
 
 ```python
-def set_head(self: ExecutionEngine, block_hash: Hash32) -> bool:
-    """
-    Returns True if the ``block_hash`` was successfully set as head of the execution payload chain.
-    """
-    ...
-```
-
-#### `finalize_block`
+HEAD_TYPE_FINALIZED_HEAD = uint64(0)
+HEAD_TYPE_SAFE_HEAD = uint64(1)
+HEAD_TYPE_UNSAFE_HEAD = uint64(2)
 
-Applies finality to the execution state: it irreversibly persists the chain of all execution payloads
-and corresponding state, up to and including `block_hash`.
-
-The body of this function is implementation dependent.
-The Consensus API may be used to implement this with an external execution engine.
-
-```python
-def finalize_block(self: ExecutionEngine, block_hash: Hash32) -> bool:
+def set_head(self: ExecutionEngine, block_hash: Hash32, head_type: unit64) -> bool:
```

Comment:
If we do want to signal head, safe_head, and finalized discretely via this interface, I suggest instead we require all pieces of data passed into this method simultaneously as to signal an *atomic* update of each. 

This is how the both API and EIP-3675 POS_EVENTS are being spec'd in practice to avoid race conditions.

I'm still mixed about the consensus specs defining "safe" here as from the consensus (and the execution EIP) perspective. To me, this seems much more of something that should just live in the API because it has no bearing on the consensus specs.

I *at least* think that this function should have a 1:1 mapping with the fields expected in `POS_FORKCHOICE_UPDATED` which right now is only head and finalized because from the perspective of the execution-layer core specs that is all that is needed

Reply:
I have started a work on updating `ExecutionEngine` interface in the spec, the goal is to have a 1:1 matching on the events and Engine API. It should be done soon.